### PR TITLE
material library reorganization and added a few materials

### PR DESCRIPTION
**Pull request by weiliangjin2021**

### CODE REVIEW

File: tidy3d/material_library/material_library.py [hunk 1]
```diff
@@ -0,0 +1,1645 @@
...
+        "that maps from a key to the variant model.",
+    )
+    default: str = pd.Field(
+        ..., title="default variant", description="The default type of variant."
+    )
+
+    @pd.validator("default", always=True)
+    def _default_in_variants(cls, val, values):
+        """Make sure the default variant is already included in the ``variants``."""
+        if not val in values["variants"].keys():
```
Comment (tylerflex) [inline tidy3d/material_library/material_library.py:50]: no need for `.keys()` here I think, just `if not val in values['variants']` already searches through keys

**Review by tylerflex — APPROVED** [review]
Just a few really minor comments. Otherwise approve! Thanks a lot for putting this together, this is a massive effort..

File: tests/test_package/test_material_library.py [hunk 2]
```diff
@@ -1,21 +1,55 @@
...
+                fmin, fmax = variant.medium.frequency_range
             else:
                 fmin, fmax = 100e12, 300e12
             freqs = np.linspace(fmin, fmax, 10011)
-            eps_complex = variant.eps_model(freqs)
-
-
-@clear_tmp
-def test_test_export():
-    export_matlib_to_file("tests/tmp/matlib.json")
```
[the remark this answers is not in the document]
Comment (tylerflex) [reply tests/test_package/test_material_library.py:21]: I think it was used mostly internally to generate a file for MC to use in the web interface.

[the remark this answers is not in the document]
Comment (tylerflex) [reply tests/test_package/test_material_library.py:21]: but with this new set up, we can just export the `material_library.to_file('matlib.json')` yes?

**Review by xin-flex — APPROVED** [review]

**Comment by xin-flex** [comment]
I added back the code to generate json file (with minor modification to work with the new format).

**Comment by tylerflex** [comment]
@xin-flex can you add back the test for the material library export?  
Also, the error is because standard library packages (like `json`) are ideally imported before user added packages (like `pydantic`). it's a bit confusing.

**Comment by xin-flex** [comment]
> @xin-flex can you add back the test for the material library export? Also, the error is because standard library packages (like `json`) are ideally imported before user added packages (like `pydantic`). it's a bit confusing.

I see, thanks! Moved the import json to the front and tests passed.
Might need @weiliangjin2021 's help to add back the test. I am not sure where/what they are.

**Comment by tylerflex** [comment]
Ok, for reference, this is what the test did before:
https://github.com/flexcompute/tidy3d/blob/61bc3a34ef362834779913f5d77cfd7e49222548/tests/test_package/test_material_library.py#L19-L21
Just basically calling that function write the material library to the temporary directory to make sure it works.

**Comment by tylerflex** [comment]
@weiliangjin2021 feel free to rebase and merge this when you think it's ready.

**Pull request merged**